py/selenium/webdriver/common/print_page_options.py

```diff
@@ -0,0 +1,154 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class PrintOptions():
+
+    def __init__(self):
+        self.orientation_values = ['portrait', 'landscape']
+        self._print_options = {}
+        self._page = {}
+        self._margin = {}
+
+    @property
+    def to_dict(self):
+        """
+        :Returns: A hash of print options configured
+        """
+        return self._print_options
+
+    def set_orientation(self, value):
+        if value not in self.orientation_values:
+            raise ValueError('Orientation value must be one of ' + str(self.orientation_values))
+
+        self._print_options['orientation'] = value
+
+    def get_orientation(self):
+        return self._print_options.get('orientation', None)
+
+    def set_scale(self, value):
+        if not isinstance(value, int) and not isinstance(value, float):
+            raise ValueError('Scale value should either be an integer or float')
+
+        if value < 0.1 or value > 2:
+            raise ValueError('Scale value should be between 0.1 and 2')
+
+        self._print_options['scale'] = value
+
+    def get_scale(self):
+        return self._print_options.get('scale', None)
+
+    def set_background(self, value):
+        if not isinstance(value, bool):
+            raise ValueError('Set background value should be a boolean')
+        self._print_options['background'] = value
+
+    def get_background(self):
+        return self._print_options.get('background', None)
+
+    def set_width(self, value):
```

Comment:
this should probably be called `page_width`, and properties that deal with `self._page` should probably start with `page_` so people understand it's use